File: VB/WinWebSolution.Module/Module.vb
Imports System
Imports DevExpress.ExpressApp
Imports System.ComponentModel
Imports DevExpress.ExpressApp.DC
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.Model


Namespace WinWebSolution.Module
    Public NotInheritable Partial Class WinWebSolutionModule
        Inherits ModuleBase

        Public Sub New()
            InitializeComponent()
        End Sub
        Public Overrides Sub ExtendModelInterfaces(ByVal extenders As DevExpress.ExpressApp.Model.ModelInterfaceExtenders)
            MyBase.ExtendModelInterfaces(extenders)
            extenders.Add(Of IModelLayoutManagerOptions, IModelOptionsHighlightFocusedLayoutItem)()
            extenders.Add(Of IModelDetailView, IModelDetailViewHighlightFocusedLayoutItem)()
        End Sub
    End Class
    Public Interface IModelOptionsHighlightFocusedLayoutItem
        Inherits IModelNode

        <DefaultValue(True), Category("Layout")> _
        Property HighlightFocusedLayoutItem() As Boolean
    End Interface
    Public Interface IModelDetailViewHighlightFocusedLayoutItem
        Inherits IModelNode

        <Category("Layout")> _
        Property HighlightFocusedLayoutItem() As Boolean
    End Interface
    <DomainLogic(GetType(IModelDetailViewHighlightFocusedLayoutItem))> _
    Public Class ModelDetailViewHighlightFocusedLayoutItemLogic
        Public Shared Function Get_HighlightFocusedLayoutItem(ByVal model As IModelDetailViewHighlightFocusedLayoutItem) As Boolean
            If model IsNot Nothing Then
                Return DirectCast(model.Application.Options.LayoutManagerOptions, IModelOptionsHighlightFocusedLayoutItem).HighlightFocusedLayoutItem
            End If
            Return False
        End Function
    End Class
End Namespace

File: CS/WinWebSolution.Module/Module.cs
using System;
using DevExpress.ExpressApp;
using System.ComponentModel;
using DevExpress.ExpressApp.DC;
using System.Collections.Generic;
using DevExpress.ExpressApp.Model;


namespace WinWebSolution.Module {
    public sealed partial class WinWebSolutionModule : ModuleBase {
        public WinWebSolutionModule() {
            InitializeComponent();
        }
        public override void ExtendModelInterfaces(DevExpress.ExpressApp.Model.ModelInterfaceExtenders extenders) {
            base.ExtendModelInterfaces(extenders);
            extenders.Add<IModelLayoutManagerOptions, IModelOptionsHighlightFocusedLayoutItem>();
            extenders.Add<IModelDetailView, IModelDetailViewHighlightFocusedLayoutItem>();
        }
    }
    public interface IModelOptionsHighlightFocusedLayoutItem : IModelNode {
        [DefaultValue(true)]
        [Category("Layout")]
        bool HighlightFocusedLayoutItem { get; set; }
    }
    public interface IModelDetailViewHighlightFocusedLayoutItem : IModelNode {
        [Category("Layout")]
        bool HighlightFocusedLayoutItem { get; set; }
    }
    [DomainLogic(typeof(IModelDetailViewHighlightFocusedLayoutItem))]
    public class ModelDetailViewHighlightFocusedLayoutItemLogic {
        public static bool Get_HighlightFocusedLayoutItem(IModelDetailViewHighlightFocusedLayoutItem model) {
            if (model != null)
                return ((IModelOptionsHighlightFocusedLayoutItem)model.Application.Options.LayoutManagerOptions).HighlightFocusedLayoutItem;
            return false;
        }
    }
}
